core/expect.ts

```diff
@@ -185,15 +185,27 @@ export class Matcher {
    }
 
    /**
-    * Checks that an array is empty
+    * Checks that an array is empty, a string is empty, or an object literal has no properties
     */
    public toBeEmpty() {
-      if (!Array.isArray(this._actualValue)) {
-         throw new TypeError("toBeEmpty requires value passed in to Expect to be an array");
+      if (null === this.actualValue || undefined === this.actualValue) {
+         throw new TypeError("toBeEmpty requires value passed in to Expect not to be null or undefined");
       }
 
-      if ((this.actualValue.length === 0) !== this.shouldMatch) {
-         throw new EmptyMatchError(this._actualValue, this.shouldMatch);
+      if (typeof this.actualValue === "string") {
```

Comment:
I did debate over joining or separating the two branches, decided to keep separate just in case the behaviour between the two types ever needed to change, if you'd prefer it in the one branch for now then I'll tweak it